src/core/annotation.js

```diff
@@ -3882,11 +3913,20 @@ class FreeTextAnnotation extends MarkupAnnotation {
     // We want to be able to add mouse listeners to the annotation.
     this.data.noHTML = false;
 
-    const { evaluatorOptions, xref } = params;
+    const { evaluatorOptions, xref, dict } = params;
     this.data.annotationType = AnnotationType.FREETEXT;
     this.setDefaultAppearance(params);
     this._hasAppearance = !!this.appearance;
 
+    if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("MOZCENTRAL")) {
+      if (dict.has("CL")) {
+        this.setLineEnding(dict.getArray("LE"));
+        this.data.calloutLine = dict.getArray("CL");
+        this.data.rectDifference = dict.getArray("RD");
```

Comment:
In a corrupt PDF document those fields can contain anything and *obviously* need to be properly parsed and validated, rather than included as-is since that could lead to failures.